picker.go

```diff
@@ -0,0 +1,56 @@
+package grpc
+
+import (
+	"time"
+)
+
+// Picker picks a Conn for RPC requests.
+// This is EXPERIMENTAL and Please do not implement your own Picker for now.
+type Picker interface {
+	// Pick returns the Conn to use for the upcoming RPC. It may return different
+	// Conn's up to the implementation.
+	Pick() (*Conn, error)
+	// State returns the connectivity state of the underlying connections.
+	State() ConnectivityState
+	// WaitForStateChange blocks until the state changes to something other than
+	// the sourceState or timeout fires on cc. It returns false if timeout fires,
+	// and true otherwise.
+	WaitForStateChange(timeout time.Duration, sourceState ConnectivityState) bool
```

Comment:
Having just a timeout here seems like a problem - what happens if the connection is closing or the caller is no longer interested in the request?

Seems like a Context (http://blog.golang.org/context) would be more appropriate.